lib/signs_ui/config/expiration.ex

```diff
@@ -59,6 +59,59 @@ defmodule SignsUi.Config.Expiration do
     |> Enum.into(%{})
   end
 
+  @spec expire_signs_via_alert(list(SignsUi.Config.State.t()), MapSet.t()) ::
+          list(SignsUi.Config.State.t())
+  def expire_signs_via_alert(sign_states, alert_ids) do
+    expired_sign_states =
+      get_expired_sign_states(
+        sign_states,
+        alert_ids,
+        []
+      )
+
+    expired_sign_states
+  end
+
+  @spec get_expired_sign_states(
+          list(SignsUi.Config.State.t()),
+          MapSet.t(),
+          list(SignsUi.Config.State.t())
+        ) :: list(SignsUi.Config.State.t())
+  def get_expired_sign_states(remaining_sign_states, alert_ids, expired_sign_states) do
+    if remaining_sign_states == [] do
+      expired_sign_states
+    else
+      [sign_state | tail] = remaining_sign_states
+
+      cond do
+        sign_state.config.mode == :auto ->
+          get_expired_sign_states(tail, alert_ids, expired_sign_states)
+
+        Enum.member?(alert_ids, sign_state.config.alert_id) ->
+          get_expired_sign_states(tail, alert_ids, expired_sign_states)
+
+        true ->
+          new_sign_state = %SignsUi.Config.Sign{
+            id: sign_state.config.alert_id,
+            config: %{mode: :auto}
+          }
+
+          expired_sign_states = [new_sign_state | expired_sign_states]
+          get_expired_sign_states(tail, alert_ids, expired_sign_states)
+      end
+    end
+  end
+
+  @spec get_active_alert_ids(SignsUi.Alerts.State.state()) ::
+          MapSet.t(String.t())
+  def get_active_alert_ids(alert_state) do
```

Comment:
In general, you don't need the `get_` prefix: the name of the function indicates what you're getting.